core/consensus/sumeragi.cpp

```diff
@@ -170,18 +171,50 @@ namespace sumeragi {
         std::string     myPublicKey;
         std::deque<std::unique_ptr<peer::Node>> validatingPeers;
 
+        Context()
+        {
+            update();
+        }
+
         Context(std::vector<std::unique_ptr<peer::Node>>&& peers)
         {
             for (auto&& p : peers) {
                 validatingPeers.push_back(std::move(p));
             }
         }
+
+        void update()
+        {
+            auto peers = config::PeerServiceConfig::getInstance().getPeerList();
+            for (auto&& p : peers ) {
+                validatingPeers.push_back( std::move(p) );
+            }
+
+            this->numValidatingPeers = this->validatingPeers.size();
+            // maxFaulty = Default to approx. 1/3 of the network.
+            this->maxFaulty = config::IrohaConfigManager::getInstance()
+                    .getMaxFaultyPeers(this->numValidatingPeers / 3);
+            this->proxyTailNdx = this->maxFaulty * 2 + 1;
+
+            if (this->validatingPeers.empty()) {
+                logger::error("sumeragi") << "could not find any validating peers.";
+                exit(EXIT_FAILURE);
+            }
+
+            if (this->proxyTailNdx >= this->validatingPeers.size()) {
+                this->proxyTailNdx = this->validatingPeers.size() - 1;
+            }
+
+            this->panicCount = 0;
+            this->myPublicKey = config::PeerServiceConfig::getInstance().getMyPublicKey();
+
+            this->isSumeragi = this->validatingPeers.at(0)->getPublicKey() == this->myPublicKey;
```

Comment:
There could be potential race conditions. Probably best not to set the isSumeragi here, but rather get this information from the peer service when starting to process.